src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -183,14 +209,25 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
         }
       }
 
-      if (!Strings.isNullOrEmpty(kmsKeyName)) {
-        loadConfig.setDestinationEncryptionConfiguration(new EncryptionConfiguration().setKmsKeyName(kmsKeyName));
-      }
+      temporaryTableReference = null;
+      if (tableExists && !isTableEmpty(tableRef) && !Operation.INSERT.equals(operation)) {
+        String temporaryTableName = UUID.randomUUID().toString().replaceAll("-", "_");
+        temporaryTableReference = new TableReference()
+          .setDatasetId(tableRef.getDatasetId())
+          .setProjectId(tableRef.getProjectId())
+          .setTableId(temporaryTableName);
+        loadConfig.setDestinationTable(temporaryTableReference);
+      } else {
+        loadConfig.setDestinationTable(tableRef);
+        if (!Strings.isNullOrEmpty(kmsKeyName)) {
+          loadConfig.setDestinationEncryptionConfiguration(new EncryptionConfiguration().setKmsKeyName(kmsKeyName));
+        }
 
-      if (allowSchemaRelaxation) {
-        loadConfig.setSchemaUpdateOptions(Arrays.asList(
-          JobInfo.SchemaUpdateOption.ALLOW_FIELD_ADDITION.name(),
-          JobInfo.SchemaUpdateOption.ALLOW_FIELD_RELAXATION.name()));
+        if (allowSchemaRelaxation) {
```

Comment:
this means schema relaxation only applies if the the operation is 'insert'. There should be a way to update the schema for the other operation types as well.

Reply:
Added possibility update table schema during Update/Upsert operation.